File: Genovationai.TaskManagement/Genovationai.TaskManagement.Core/Entities/Task.cs


using Genovationai.TaskManagement.Core.Abstraction;

namespace Genovationai.TaskManagement.Core.Entities;

/// <summary>
/// Represents a task that can be assigned to a team member.
/// </summary>
public class Task : BaseEntity
{
    public string Title { get; set; } = null!;

    public string? Description { get; set; }

    public int? AssignedToId { get; set; }

    public TaskStatus Status { get; set; } = TaskStatus.NotStarted;

    public TeamMember? AssignedTo { get; set; }
}

public enum TaskStatus : int
{
    NotStarted = 1,
    InProgress = 2,
    Completed = 3,
    OnHold = 4,
    Cancelled = 5
}

File: task-management-frontend/src/pages/edit-task/task.ts


export interface Task {
  id: number;
  title: string;
  description: string;
  status: string;
  assignedToId?: number;
  assignedToFirstName?: string;
  assignedToLastName?: string;
}


export enum TaskStatus {
  NotStarted = 1,
  InProgress = 2,
  Completed = 3,
  OnHold = 4,
  Cancelled = 5
}